src/main/java/no/unit/nva/institution/proxy/handler/NestedInstitutionHandler.java

```diff
@@ -30,6 +33,7 @@
     public static final String PARAMETER_NOT_FOUND_ERROR_MESSAGE = "Parameter not found:";
 
     private final CristinApiClient cristinApiClient;
+    private final Map<String, JsonNode> cache = new ConcurrentHashMap<>();
```

Comment:
Have you considered to cache in CloudFront?

https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/QueryStringParameters.html

Reply:
Two problems with this — cloudfront would need a response to actually cache — the specific use case here is that the response has been processed, but API Gateway times out before the response is generated. Secondly, the caching of the query in cloudfront may cause problems because this is in fact a query…if the Cristin data changes.

Reply:
I think this cache is going to survive only until this process dies, so given that the process is going to die after some minutes, I am not sure how much it is going to be used. It does not harm however. Maybe we should put a limit on how much this cache should be so that we do run into an OutOfMemoryException. Maybe we could create later a more sophisticated cache using DynamoDB.